src/Symfony/Component/AssetMapper/ImportMap/ImportMapAuditor.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace Symfony\Component\AssetMapper\ImportMap;
+
+use Symfony\Component\AssetMapper\Exception\RuntimeException;
+use Symfony\Component\HttpClient\HttpClient;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+class ImportMapAuditor
+{
+    private const AUDIT_URL = 'https://registry.npmjs.org/-/npm/v1/security/audits';
+
+    private readonly HttpClientInterface $httpClient;
+
+    public function __construct(
+        private readonly string $importMapConfigPath,
+        HttpClientInterface $httpClient = null,
+    ) {
+        $this->httpClient = $httpClient ?? HttpClient::create();
+    }
+
+    public function audit(): array
+    {
+        $packages = $this->getPackagesVersion();
+
+        if ([] === $packages) {
+            return [];
+        }
+
+        $response = $this->httpClient->request('POST', self::AUDIT_URL, [
+            'json' => [
+                'requires' => array_map(fn(): string => '*', $packages),
+                'dependencies' => array_map(fn(string $version): array => ['version' => $version], $packages),
+            ],
+        ]);
+
+        if (200 !== $response->getStatusCode()) {
+            throw new RuntimeException(sprintf('Error %d auditing packages. Response: %s', $response->getStatusCode(), $response->getContent(false)));
+        }
+
+
+        $data = $response->toArray();
+
+        return [
+            'packages' => $packages,
+            'advisories' => $data['advisories'],
+            'summary' => $data['metadata']['vulnerabilities'],
+        ];
+    }
+
+    private function getPackagesVersion(): array
+    {
+        $path = $this->importMapConfigPath;
+        $importMapConfig = is_file($path) ? (static fn () => include $path)() : [];
+
+        $packages = [];
+        foreach ($importMapConfig as $data) {
+            if (null === $url = $data['url'] ?? null) {
+                continue;
+            }
+
+            // try to grab the version from the URL
+            $registryRegex = '(?<registry>\w+)';
+            $packageRegex = '(?<package>(?:@[a-z0-9-~][a-z0-9-._~]*/)?[a-z0-9-~][a-z0-9-._~]*)';
+            $versionRegex = '(?<version>[\w\._-]+)';
+            $subPathRegex = '(?<subpath>/.*)';
+
+            if (
+                1 === preg_match("#^https://ga.jspm.io/{$registryRegex}:{$packageRegex}@{$versionRegex}{$subPathRegex}?$#", $url, $matches)
```

Comment:
Let's extract this section to the individual [PackageResolverInterface](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/AssetMapper/ImportMap/Resolver/PackageResolverInterface.php) classes - e.g. a new `getPackageInfo(string $url): ?array` which would return... an array with `package` and `version` keys or `null` of a resolver doesn't support the URL (or maybe we make a new small class instead of the array).

We'll need to reuse this logic for an `asset-map:outdated` command. So putting it in the package resolvers would allow that.

Reply:
Given that the interface is an extension point of the package, I suggest using a value object rather than an array shape.

Reply:
Maybe we can add this check at "resolvePackages" step, add a `?string $version` property to `ResolvedImportMapPackage` and dump it in `importmap.php` file. It will look like:
```php
<?php

/*
 * This file is part of the Symfony package.
 *
 * (c) Fabien Potencier <fabien@symfony.com>
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 */

return [
    '@hotwired/stimulus' => [
        'url' => 'https://unpkg.com/@hotwired/stimulus@3.2.1/dist/stimulus.js',
    ],
    'json5' => [
        'url' => 'https://cdn.jsdelivr.net/npm/json5@1.0.0/+esm',
    ],
    'lodash' => [
        'url' => 'https://ga.jspm.io/npm:lodash@4.17.21/lodash.js',
    ],
    'file' => [
        'path' => 'some/file.js',
    ],
];

Reply:
You can review this in the last commit.